### PR TITLE
kernelci/config: add email configuration to kernelci-core

### PR DESCRIPTION
**Pull request by alexandrasp**
Include a class in KernelCI core to load a basic set of email configurations.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

### CODE REVIEW

**Comment by mgalka** [comment]
@alexandrasp Please, add a comment about testing

**Comment by mgalka** [comment]
@alexandrasp regarding the commit message 

```
kernelci/config: add email configurations to kernelci-core
include a class in KernelCI core to load a basic set
of email configurations.
```

I think it should be "configuration" rather than "configurations"

**Comment by alexandrasp** [comment]
> Include a class in KernelCI core to load basic e-mail configuration.

Sure, updated ;)

**Comment by mgalka** [comment]
> > Include a class in KernelCI core to load basic e-mail configuration.
> 
> Sure, updated ;)

The commit header still says "add email configurations"

File: kernelci/config/__init__.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
 import kernelci.config.lab
 import kernelci.config.rootfs
 import kernelci.config.test
+import kernelci.config.email
```
Comment (gctucker) [inline kernelci/config/__init__.py:29]: Please keep in alphabetical order

File: kernelci/config/__init__.py [hunk 2]
```diff
@@ -93,6 +94,7 @@ def from_data(data):
     config.update(kernelci.config.lab.from_yaml(data, filters))
     config.update(kernelci.config.rootfs.from_yaml(data, filters))
     config.update(kernelci.config.test.from_yaml(data, filters))
+    config.update(kernelci.config.email.from_yaml(data))
```
Comment (gctucker) [inline kernelci/config/__init__.py:97]: Please keep in alphabetical order

File: kernelci/config/email.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    def email_send_to(self):
+        return self._email_send_to
+
+    @property
+    def email_send_from(self):
+        return self._email_send_from
+
+    @property
+    def email_subject(self):
+        return self._email_subject
```
Comment (gctucker) [inline kernelci/config/email.py:49]: The subject shouldn't be part of the config as it should be set by the template and the rendering code.

File: kernelci/config/email.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    def email_type(self):
+        return self._email_type
+
+    @property
+    def email_send_to(self):
+        return self._email_send_to
+
+    @property
+    def email_send_from(self):
+        return self._email_send_from
```
Comment (gctucker) [inline kernelci/config/email.py:45]: I don't think the "from" field should be configurable in YAML, it's a system-wide parameter like the SMTP port etc.

File: kernelci/config/email.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        self._email_send_from = email_send_from
+        self._email_subject = email_subject
+
+    @classmethod
+    def from_yaml(cls, email_type, kw):
+        return cls(email_type, **kw)
+
+    @property
+    def email_type(self):
+        return self._email_type
```
Comment (gctucker) [inline kernelci/config/email.py:37]: What is the email "type" supposed to be?

**Review by gctucker — COMMENTED** [review]
I think that while we will need some YAML configuration, this hasn't been completely thought through so there are a few things that will need to be redesigned.

File: kernelci/config/email.py (hunk 4 above)
Comment (alexandrasp) [reply kernelci/config/email.py]: It makes sense. I removed subject as well.. at this moment not sure about what we should have in this YAML other than recipients list and the environment. Should I consider putting anything else?

File: kernelci/config/__init__.py (hunk 1 above)
Comment (alexandrasp) [reply kernelci/config/__init__.py]: updated

File: kernelci/config/__init__.py (hunk 2 above)
Comment (alexandrasp) [reply kernelci/config/__init__.py]: updated

File: kernelci/config/email.py (hunk 3 above)
Comment (mgalka) [reply kernelci/config/email.py]: @gctucker Do you have any thoughts on how this value should be provided to the script which renders the e-mail from template?

Comment (alexandrasp) [reply kernelci/config/email.py]: @mgalka maybe could be like this: https://github.com/kernelci/kernelci-pipeline/pull/72/files#diff-a356af3cdd34dc1001fadc4e67566b93959d3e9c1d50946c3032112b186557d0R149 as in the legacy back-end.

**Comment by nuclearcat** [comment]
ping, is any work planned on this PR or it will be made as new PR?

**Comment by gctucker** [comment]
We do need YAML-based email configuration and code to handle that and replace `kernel-arch-complete.sh`, but this PR is not implementing the actual features we need.  So I think we can just close it.

**Comment by nuclearcat** [comment]
Thank you, closing

**Pull request closed without merging**